Duplicati/Library/Main/Operation/RestoreHandler.cs

```diff
@@ -415,11 +415,21 @@ private void DoRun(LocalDatabase dbparent, Library.Utility.IFilter filter, Resto
                 // Restore empty files. They might not have any blocks so don't appear in any volume.
                 foreach (var file in database.GetFilesToRestore(true).Where(item => item.Length == 0))
                 {
-                    SystemIO.IO_OS.DirectoryCreate(SystemIO.IO_OS.PathGetDirectoryName(file.Path));
+                    Logging.Log.WriteProfilingMessage(LOGTAG, "RestoreFile", "restoring empty file \"{0}\"", file.Path);
 
-                    // Just create the file and close it right away, empty statement is intentional.
-                    using (SystemIO.IO_OS.FileCreate(file.Path))
+                    try
+                    {
+                        SystemIO.IO_OS.DirectoryCreate(SystemIO.IO_OS.PathGetDirectoryName(file.Path));
+                        // Just create the file and close it right away, empty statement is intentional.
+                        using (SystemIO.IO_OS.FileCreate(file.Path))
+                        {
+                        }
+                    }
+                    catch (Exception ex)
                     {
+                        Logging.Log.WriteWarningMessage(LOGTAG, "RestoreFileFailed", ex, "Failed to restore empty file: \"{0}\", message: {1}", file.Path, ex.Message);

```

Comment:
For non-empty files, we write an error message, and also increment the `fileErrors` variable.  Should we do the same here?

https://github.com/duplicati/duplicati/blob/9b977b7ba04a940868754fd15a37b99fbd77f1bd/Duplicati/Library/Main/Operation/RestoreHandler.cs#L472-L478